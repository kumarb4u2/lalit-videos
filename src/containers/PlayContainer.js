import React from 'react';
import { withRouter } from 'react-router-dom';
import styled from 'styled-components';
import { BASE_URL } from '../helper/constants';

const VideoContainer = styled.div`
  margin-top: 30px;
  text-align: center;
`;

const Heading = styled.h2`
  color: blue;
`;

const PlayContainer = props => (
  <div className="container-padding">
    <Heading>Playing {props.match.params.id}</Heading>
    <div className="row justify-content-sm-center">
      <VideoContainer className="col-xs-12 col-sm-6">
        <video width="100%" controls autoPlay>
          <source
            src={`${BASE_URL}/uploads/${props.match.params.id}`}
            type="video/mp4"
          />
          Your browser does not support the video tag.
        </video>
      </VideoContainer>
    </div>
  </div>
);

export default withRouter(PlayContainer);
