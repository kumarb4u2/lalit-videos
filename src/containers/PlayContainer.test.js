import React from 'react';
import { shallow } from 'enzyme';
import PlayContainer from './PlayContainer';

describe('PlayContainer', () => {
  it('should match snapshot in default state', () => {
    const component = shallow(<PlayContainer />);
    expect(component).toMatchSnapshot();
  });
  it('should match snapshot if video id is passed', () => {
    const match = { params: { id: '1231321' } };
    const component = shallow(<PlayContainer match={match} />);
    expect(component).toMatchSnapshot();
  });
});
