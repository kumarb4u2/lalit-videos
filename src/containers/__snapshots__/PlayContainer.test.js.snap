// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayContainer should match snapshot if video id is passed 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(PlayContainer)
    match={
      Object {
        "params": Object {
          "id": "1231321",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`PlayContainer should match snapshot in default state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(PlayContainer) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
